controller/apps.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if env == nil {
+		return errors.NewNotFoundError("environment", ctx.EnvName)
+	}
+
+	res := &app.SimpleEnvironmentSingle{
+		Data: env,
+	}
+
+	// AppsController_ShowEnvironment: end_implement
+	return ctx.OK(res)
+}
+
+// ShowSpace runs the showSpace action.
+func (c *AppsController) ShowSpace(ctx *app.ShowSpaceAppsContext) error {
+	// AppsController_ShowSpace: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeSpaceName, err := c.getSpaceNameFromSpaceID(ctx, ctx.SpaceID)
+	if err != nil {
+		return err
+	}
+
+	// get OpenShift space
+	space, err := kc.GetSpace(*kubeSpaceName)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if space == nil {
+		return errors.NewNotFoundError("space", *kubeSpaceName)
+	}
+
+	res := &app.SimpleSpaceSingle{
+		Data: space,
+	}
+
+	// AppsController_ShowSpace: end_implement
+	return ctx.OK(res)
+}
+
+// ShowSpaceApp runs the showSpaceApp action.
+func (c *AppsController) ShowSpaceApp(ctx *app.ShowSpaceAppAppsContext) error {
+	// AppsController_ShowSpaceApp: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeSpaceName, err := c.getSpaceNameFromSpaceID(ctx, ctx.SpaceID)
+	if err != nil {
+		return err
+	}
+
+	theapp, err := kc.GetApplication(*kubeSpaceName, ctx.AppName)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if theapp == nil {
+		return errors.NewNotFoundError("application", ctx.AppName)
+	}
+
+	res := &app.SimpleApplicationSingle{
+		Data: theapp,
+	}
+
+	// AppsController_ShowSpaceApp: end_implement
+	return ctx.OK(res)
+}
+
+// ShowSpaceAppDeployment runs the showSpaceAppDeployment action.
+func (c *AppsController) ShowSpaceAppDeployment(ctx *app.ShowSpaceAppDeploymentAppsContext) error {
+	// AppsController_ShowSpaceAppDeployment: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeSpaceName, err := c.getSpaceNameFromSpaceID(ctx, ctx.SpaceID)
+	if err != nil {
+		return err
+	}
+
+	deploymentStats, err := kc.GetDeployment(*kubeSpaceName, ctx.AppName, ctx.DeployName)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if deploymentStats == nil {
+		return errors.NewNotFoundError("deployment statistics", ctx.DeployName)
+	}
+
+	res := &app.SimpleDeploymentSingle{
+		Data: deploymentStats,
+	}
+
+	// AppsController_ShowSpaceAppDeployment: end_implement
+	return ctx.OK(res)
+}
+
+// ShowEnvAppPods runs the showEnvAppPods action.
+func (c *AppsController) ShowEnvAppPods(ctx *app.ShowEnvAppPodsAppsContext) error {
+	// AppsController_ShowEnvAppPods: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	pods, err := kc.GetPodsInNamespace(ctx.EnvName, ctx.AppName)
+	if err != nil {
+		return err
+	}
+
+	jsonresp := "{\"pods\":" + tostring(pods) + "}\n"
```

Comment:
you may prefer declaring a type in the Goa `design` stage and use it here (Goa will take care of the JSON marshalling)

Reply:
I wanted to do this, but I don't know enough about Goa to convince it to use simple Go structs, or Goa types defined elsewhere.  The syntax is probably pretty simple, but I had no luck.